src/Cardano/Wallet/Api/Types/Amount.hs

```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+module Cardano.Wallet.Api.Types.Amount
+    ( Amount (..)
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Api.JSON
+    ( defaultRecordTypeOptions )
+import Data.Aeson
+    ( FromJSON (..), ToJSON (..), genericParseJSON, genericToJSON )
+import GHC.Generics
+    ( Generic )
+import Numeric.Natural
+    ( Natural )
+
+import qualified Cardano.Wallet.Api.Types.CurrencyUnit as T
+
+data Amount = Amount
+    { _quantity :: Natural
+    , _unit :: T.CurrencyUnit
+    } deriving (Eq, Generic, Show)
```

Comment:
For every types containing a unit, I'll suggest to re-use the approach of the old code base:

-> https://github.com/input-output-hk/cardano-sl/blob/1d2b72f652cbc89691c60a4d6804fc8116d7678c/lib/src/Pos/Util/UnitsOfMeasure.hs

Minus the `Buildable` and `ToSwagger` instances, this code is quite concise and allows us to group the logic of representing types with units.